components/style/stylist.rs

```diff
@@ -1065,6 +1065,21 @@ impl Stylist {
                                      CascadeFlags::empty(),
                                      self.quirks_mode))
     }
+
+    /// Accessor for a shared reference to the device.
+    pub fn device(&self) -> &Arc<Device> {
+        &self.device
+    }
+
+    /// Accessor for a mutable reference to the device.
+    pub fn device_mut(&mut self) -> &mut Arc<Device> {
+        &mut self.device
```

Comment:
I'm not sure it's possible to return `&mut Device`, right? I think we could make this work using [`OwningRef`](https://github.com/Kimundi/owning-ref-rs). This would return an `OwningRefMut<Arc<Device>, Device>`, I think.

I don't know if it's worth the churn though, so perhaps this is fine for now.